### PR TITLE
Extract report generation to reports package

### PR DESCRIPTION
**Pull request by charlieegan3**
This will allow me to generate a cluster summary and update that too
when writing output to a bucket destination.

Before this, the report generation all happened in the json exporter
which meant the report data structure wasn't available for reuse.

Done as part of https://github.com/jetstack/preflight/issues/54

Related to https://github.com/jetstack/preflight/issues/1

### CODE REVIEW

**Comment by wwwil** [comment]
Looking over this has reminded me that the other output formats are generated directly from results and a policy manifest. I feel it would be better to actually produce a report, then generate other formats of output from that. Though that can be dealt with in a separate issue.

File: pkg/reports/reports.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import (
+	"github.com/jetstack/preflight/api"
+	"github.com/jetstack/preflight/pkg/packaging"
+	"github.com/jetstack/preflight/pkg/results"
+	"github.com/jetstack/preflight/pkg/rules"
+	"github.com/jetstack/preflight/pkg/version"
+)
+
+// ConstructReport creates a report from a policy manifest and a results collection
+func ConstructReport(pm *packaging.PolicyManifest, rc *results.ResultCollection) (api.Report, error) {
```
Comment (wwwil) [inline pkg/reports/reports.go]: Is there any reason this is called `ConstructReport` rather than `Create Report`?

Comment (charlieegan3) [reply pkg/reports/reports.go]: I guess I'm seeing this as a replacement for an oop constructor method.

create works too.

I'm wondering if `NewReport` might be more idiomatic :man_shrugging:

Comment (charlieegan3) [reply pkg/reports/reports.go]: I've renamed it to new in the upcoming changes I was preparing to add on top here https://github.com/charlieegan3/preflight/pull/2

Comment (charlieegan3) [reply pkg/reports/reports.go]: Brought that in for this branch here: b47c45d70086844584894c93a1f17fc0d318a8a1

**Comment by charlieegan3** [comment]
Thanks Wil.

**Pull request merged**